Training/Training/test.c

```diff
@@ -0,0 +1,132 @@
+#pragma warning(disable:6385)
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------
+// test.c
+// Tests for selection sort and binary search
+// Sudarson S
+// ------------------------------------------------------------------------------------------------
+#include"training.h"
+#include<stdbool.h>
+#include<stdio.h>
+#include<string.h> // USED
+#include<errno.h>
+#include<malloc.h>
+#include<stdlib.h>
+#include <time.h>
+
+/// <summary>Returns user inputs only if they are valid integers</summary>
+static int GetValidInteger (char* prompt) {
+   bool output;
+   int num;
+   do {
+      char numberStr[4096], * result = NULL;
+      printf ("%s", prompt);
+      fgets (numberStr, sizeof (numberStr), stdin);
+      result = strchr (numberStr, '\n');
+      if (result == NULL) {
+         while (getchar () != '\n');
+         output = false;
+      } else {
+         errno = 0;
+         char* endptr = NULL;
+         num = strtol (numberStr, &endptr, 10);
+         output = endptr != numberStr && *endptr == '\n' && errno != ERANGE && numberStr[0] != ' ' &&
+            numberStr[0] != '\t';
+      }
+      if (!output) printf ("Invalid Input\n");
+   } while (output == false);

```

Comment:
```suggestion
   } while (!output);

```
Always check your code to see if there are other places certain changes are applicable.